src/groups/mqb/mqbstat/mqbstat_queuestats.t.cpp

```diff
@@ -490,6 +495,171 @@ static void test4_queueStatsDomainContent()
+                                 isCSL,
+                                 isFSM,
+                                 clusterNodeDefs);
+    mqbmock::Domain                mockDomain(&mockCluster, s_allocator_p);
+
+    // Reconfigure the domain with appIds and enabled appId metrics
+    const char* k_APPID_FOO = "foo";
+    const char* k_APPID_BAR = "bar";
+    const char* k_APPID_BAZ = "baz";
+
+    mqbconfm::Domain           domainConfig(s_allocator_p);
+    mqbconfm::QueueModeFanout& mode = domainConfig.mode().makeFanout();
+    mode.publishAppIdMetrics()      = true;
+    mode.appIDs().push_back(k_APPID_FOO);
+
+    mwcu::MemOutStream errorDesc(s_allocator_p);
+    mockDomain.configure(errorDesc, domainConfig);
+
+    // Do not use stat context (`mockDomain.queueStatContext()`) declared
+    // within mock domain, since it was not initialized using the proper
+    // config.  Init a new stats object from the scratch instead.
+    mqbstat::QueueStatsDomain stats(s_allocator_p);
+    stats.initialize(bmqt::Uri("bmq://mock-domain/abc", s_allocator_p),
+                     &mockDomain);
+
+    mwcst::StatContext* sc = stats.statContext();
+
+    // Make a snapshot to get a recent update with a newly initialized
+    // subcontext for "foo"
+    {
+        sc->snapshot();
+        ASSERT_EQ(1, sc->numSubcontexts());
+
+        const mwcst::StatContext* fooSc = sc->getSubcontext(k_APPID_FOO);
+        ASSERT_NE(bsl::nullptr_t(), fooSc);
+    }
+
+    // Add event for non-configured appId "bar", this value should not reach to
+    // the final stats
+    stats.onEvent(mqbstat::QueueStatsDomain::EventType::e_CONFIRM_TIME,
+                  1000,
+                  k_APPID_BAR);
+
+    // Reconfigure queue domain stats, by excluding "foo" and including "bar"
+    // and "baz"
+    {
+        bsl::vector<bsl::string> appIds(s_allocator_p);
+        appIds.push_back(k_APPID_BAR);
+        appIds.push_back(k_APPID_BAZ);
+
+        stats.updateDomainAppIds(appIds);
+
+        sc->snapshot();
+        sc->cleanup();
+
+        ASSERT_EQ(2, sc->numSubcontexts());
+
+        const mwcst::StatContext* fooSc = sc->getSubcontext(k_APPID_FOO);
+        ASSERT_EQ(bsl::nullptr_t(), fooSc);
+
+        const mwcst::StatContext* barSc = sc->getSubcontext(k_APPID_BAR);
+        const mwcst::StatContext* bazSc = sc->getSubcontext(k_APPID_BAZ);
+        ASSERT_NE(bsl::nullptr_t(), barSc);
+        ASSERT_NE(bsl::nullptr_t(), bazSc);
+    }
+
+    // Report some metrics and check that they reached subcontexts
+    {
+        stats.onEvent(mqbstat::QueueStatsDomain::EventType::e_CONFIRM_TIME,
+                      700,
+                      k_APPID_BAR);
+        stats.onEvent(mqbstat::QueueStatsDomain::EventType::e_CONFIRM_TIME,
+                      900,
+                      k_APPID_BAR);
+        stats.onEvent(mqbstat::QueueStatsDomain::EventType::e_CONFIRM_TIME,
+                      800,
+                      k_APPID_BAR);
+
```

Comment:
```suggestion
```

Reply:
The empty line here is intended, it visually separates BAR and BAZ appIds